drivers/net/phy/qsfp-mem-core.c

```diff
@@ -159,50 +147,17 @@ static void qsfp_check_hotplug(struct work_struct *work)
 	schedule_delayed_work(&qsfp->dwork, msecs_to_jiffies(QSFP_CHECK_TIME));
 }
 
-static ssize_t qsfp_connected_show(struct device *dev,
-				   struct device_attribute *attr, char *buf)
+int qsfp_init_work(struct qsfp *qsfp)
 {
-	struct qsfp *qsfp = dev_get_drvdata(dev);
-	u32 plugin;
-
-	mutex_lock(&qsfp->lock);
-	plugin = check_qsfp_plugin(qsfp) && (qsfp->init == QSFP_INIT_DONE);
-	mutex_unlock(&qsfp->lock);
-
-	return sysfs_emit(buf, "%u\n", plugin);
+	INIT_DELAYED_WORK(&qsfp->dwork, qsfp_check_hotplug);
+	qsfp_check_hotplug(&qsfp->dwork.work);
+	return 0;
 }
+EXPORT_SYMBOL_GPL(qsfp_init_work);
 
-static DEVICE_ATTR_RO(qsfp_connected);
-
-static struct attribute *qsfp_mem_attrs[] = {
-	&dev_attr_qsfp_connected.attr,
-	NULL,
-};
-ATTRIBUTE_GROUPS(qsfp_mem);
-
-static int qsfp_probe(struct dfl_device *dfl_dev)
+int qsfp_register_regmap(struct qsfp *qsfp)
 {
-	struct device *dev = &dfl_dev->dev;
-	struct qsfp *qsfp;
-#if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0) && RHEL_RELEASE_CODE < 0x803
-	int ret;
-#endif
-
-	qsfp = devm_kzalloc(dev, sizeof(*qsfp), GFP_KERNEL);
-	if (!qsfp)
-		return -ENOMEM;
-
-	qsfp->base = devm_ioremap_resource(dev, &dfl_dev->mmio_res);
-	if (!qsfp->base)
-		return -ENOMEM;
-
-	qsfp->dev = dev;
-	mutex_init(&qsfp->lock);
-
-	dev_set_drvdata(dev, qsfp);
-
-	INIT_DELAYED_WORK(&qsfp->dwork, qsfp_check_hotplug);
-	qsfp_check_hotplug(&qsfp->dwork.work);
+	struct device *dev = qsfp->dev;
 
 #if LINUX_VERSION_CODE < KERNEL_VERSION(5, 4, 0) && RHEL_RELEASE_CODE < 0x803
```

Comment:
The backport to kernel before 5.4, that is CentOS 8.2 does not compile and does not run. This should be moved to the file qsfp-mem-dfl.c in the probe function. qsfp_mem_groups is defined (using a macro) in that file.